erpnext/accounts/report/statement_of_account/statement_of_account.py

```diff
@@ -267,10 +275,52 @@ def get_addresses(company=None, party_type=None, party=None):
 	if not (company and party_type and party):
 		return {}
 
-	company_addr = frappe.get_doc("Address", get_default_address("Company", company))
-	party_addr = frappe.get_doc("Address", get_default_address(party_type, party))
+	company_address = frappe.get_doc("Address", get_default_address("Company", company))
+	party_address = frappe.get_doc("Address", get_default_address(party_type, party))
+
+	if not company_address:
+		frappe.throw(_("Please add default primary company address"))
+
+	if not party_address:
+		frappe.throw(_("Please add default primary customer address"))
```

Comment:
- This validation is not working, I can dismiss the frappe.throw and click on email report
- ![image](https://user-images.githubusercontent.com/7310479/97832516-03231f80-1cf9-11eb-8db9-59f827328797.png)
- ![image](https://user-images.githubusercontent.com/7310479/97832534-0f0ee180-1cf9-11eb-8c2f-34fc87bb2f8f.png)